pages/agent/v3/signed_pipelines.md

```diff
@@ -0,0 +1,130 @@
+# Signed pipelines
+
+> ⚠️ Warning: Signed Pipelines are experimental, and these docs (and the commands and arguments referenced in them) are subject to change. This is also an opt-in feature - if you're keen to get your hands on it, talk to your sales rep.
+
+## What are signed pipelines?
+
+Signed Pipelines is a security feature on the Buildkite Agent that allows you to cryptographically sign steps that are uploaded to Buildkite, and then verify that signature on the Buildkite Agent before running the step. This signature helps agents establish the provenance of the jobs that they are instructed to run - asserting that the steps were uploaded by a trusted agent, and that they haven't been modified since they were uploaded. If an agent detects a signature mismatch, by default, it'll refuse to run the job.
+
+These signatures mean that if a threat actor were able to modify a job in flight, the Agent would refuse to run it due to mismatched signatures.
+
+<details>
+  <summary>I think I've seen this before...</summary>
+  This work is inspired by the older <a href="https://github.com/buildkite/buildkite-signed-pipeline"><code>buildkite-signed-pipeline</code></a> tool, which was a tool you could add to your agent instances. It had a similar idea - signing steps before they're uploaded to Buildkite, then verifying them when they're run. However, it had a number of limitations, including:
+  <ul>
+    <li>It had to be installed on every agent instance, leading to more configuration</li>
+    <li>It only supported symmetric HS256 signatures, meaning that every verifier could also sign uploads</li>
+    <li>It couldn't sign matrix steps</li>
+  </ul>
+
+  This newer version of pipeline signing is built right into the agent, and addresses all of these limitations. Being built into the agent, it's easier to configure and use.
+
+  Many thanks to Seek.com.au, who we collaborated with on the older version of the tool, and whose prior art has been instrumental in the development of this newer version.
+</details>
+
+### What we actually sign
+
+As mentioned above, Signed Pipelines establishes that important aspects of steps haven't been changed since pipeline upload. In more detail, we sign the following aspects of steps:
+
+- The command
+- Environment variables _as defined in the pipeline YAML_ - environment variables set by the agent, by hooks, or by the user's shell are not signed and can potentially override the environment that a step's command is started with.
+
+- Any plugins used by the step, and their configurations
+- The matrix configuration
+  + The matrix configuration is signed as a whole, rather than each individual matrix job. This means that the signature will be the same for each job in the matrix. When signatures are verified for matrix jobs, we double-check that the job we've received is a valid construction of the matrix, and that the signature matches the matrix configuration.
+- The step's key, if present [WIP]
```

Comment:
What's the go with everything labelled "WIP?" Is it to communicate what work is coming next? If so, what's the timeline you're looking at and how confident are we in it?

(I would generally avoid documenting anything "coming soon," but worth considering in the context if you think it's important)